docs/src/main/asciidoc/config-yaml.adoc

```diff
@@ -109,55 +210,116 @@ app:
 ----
 
-.application-staging.yaml
+.`application-dev.yaml`
 [source, yaml]
 ----
 quarkus:
   http:
     port: 8082
 ----
 
-== Expressions
+// meta:include::modules/proc_setting-custom-configuration-profiles-with-yaml.adoc[leveloffset=+2]
+[id="proc_setting-custom-configuration-profiles-with-yaml_{context}"]
+== Setting custom configuration profiles with YAML
 
-The YAML format also supports xref:config-reference.adoc#property-expressions[property expressions], using the same format as Java
-Properties:
+With {project-name}, you can set configuration properties and values specific to your application's different configuration profiles.
+You can start your application with a specific profile to access a particular configuration.
+This procedure shows how you can configure a specific profile in YAML format.
 
-[source, yaml]
+[TIP]
+====
+In YAML, you must place all strings that begin with a special character inside quotation marks.
+====
+
+.Prerequisites
+
+include::{includes}/prerequisites.adoc[]
+* A {project-name} Maven project configured to use a PostgreSQL data source with a JDBC data source driver.
+* The `quarkus-jdbc-postgresql` and `quarkus-config-yaml` extension as dependencies in your project's `pom.xml` or `build.gradle` file.
+
+.Procedure
+
+. Open the `src/main/resources/application.yaml` configuration file.
+
+. To add a profile-specific configuration, enclose the profile's configuration key between double quotes, and then list the key-value pairs.
++
+.`src/main/resources/application.yaml`
+[source,yaml]
+----
+"%dev":
+  # Properties that configure the JDBC data source driver of your PostgreSQL data source
+  quarkus:
+    datasource:
+      url: jdbc:postgresql://localhost:5432/some-database
+      driver: org.postgresql.Driver
+      username: quarkus
+      password: quarkus
+----
+
+// WAS meta:include::modules/con_property-expressions.adoc[leveloffset=+1]
+// NEW meta:include::modules/proc_using-property-expressions.adoc[leveloffset=+1]
+[id="con_property-expressions_{context}"]
+== Using property expressions in a YAML file
+
+Property expressions are similar to variables: instead of hardcoding a property value, you can use a property expression.
+The the configuration system resolves the property expression when it gets the values from the configuration source in the application:
+
+* If it reads the configuration property from your configuration at compile time, the property expression is also resolved at compile time.
+* If the configuration property is overridden at run time, its value is resolved at run time.
+
+You can use multiple sources to resolve property expressions.
+You can use a property value from one source to expand an expression in another.
+
+If the value of a property in an expression cannot be resolved, and you did not set a default value for the expression, your application encounters a `NoSuchElementException`.
+
+.Procedure
+
+Property expressions have the `${<property_name>}` format.
```

Comment:
I would probably omit the `<` and the `>`, they may confuse the user as being part of the syntax. Alternative, we can add something like: 

>  where `<property_name>` corresponds to the configuration name to expand


>

Reply:
Removed. Thanks.